### PR TITLE
Add command function for shell surface to set position

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #419.

This is my (WIP) PR for a command for setting the position of a
(shell) surface.

The actual implementation of the command _should_ be ready, but I
did not yet add code for setting the position of a shell surface
by now, as I think I need help from @TheNeikos for this.

Please do a short review if things look right by now
(especially @TheNeikos).

### CODE REVIEW

File: src/compositor/wayland/shell_surface.c [hunk 1]
```diff
@@ -381,3 +423,32 @@ resource_destroy(
...
+) {
+    //!< @todo implement
+    return 0;
+}
+
+static int
+cmd_func_set_pos(
+    union ws_value_union* stack
+) {
+    if (ws_value_get_type(&stack[0].value) != WS_VALUE_TYPE_OBJECT_ID ||
```
Comment (matthiasbeyer) [inline src/compositor/wayland/shell_surface.c:427]: This must be `2`

Comment (neithernut) [inline src/compositor/wayland/shell_surface.c:427]: Nope, he checks whether the first two arguments are sane here.

Comment (matthiasbeyer) [inline src/compositor/wayland/shell_surface.c:427]: "He" is me btw.

I guess this function gets the context and function name as first arguments, too, so I incremented the stack ptr by hand here (this diff is outdated).

File: src/compositor/wayland/shell_surface.c [hunk 2]
```diff
@@ -169,6 +175,35 @@ resource_destroy(
...
+ * @memberof ws_shell_surface
+ *
+ * @return zero on success, else negative errno.h number
+ */
+static int
+move(
+    struct wl_resource* resource, //!< The shell surface
+    int32_t x, //!< The requested X position
+    int32_t y //!< The requested Y position
+);
```
Comment (neithernut) [inline src/compositor/wayland/shell_surface.c]: Please make this a public method.

File: src/compositor/wayland/shell_surface.h [hunk 3]
```diff
@@ -65,5 +65,19 @@ ws_shell_surface_new(
...
+ * @memberof ws_shell_surface
+ *
+ * @return zero on success, else negative errno.h number
+ */
+int
+ws_shell_surface_move(
+    struct wl_resource* resource,
+    int32_t x,
+    int32_t y
+);
```
Comment (neithernut) [inline src/compositor/wayland/shell_surface.h:80]: It's unclear if this moves the sufacerelatively or whether it sets an absolute positon, given only the method name. Please rename to something more intuitive (like `ws_shell_set_pos`).

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**